graftm/decorator.py

```diff
@@ -0,0 +1,83 @@
+
+# System imports
+import logging
+from skbio import TreeNode
+from graftm.reannotator import Reannotator
+from graftm.tree_decorator import TreeDecorator
+
+class Decorator:
+    '''re-root a tree and decorate it using old taxonomy, for the graftM 
+    decorate pipeline.
+    
+    ALL trees provided to this class must be in newick format. No other formats 
+    are supported. If any taxonomic decoration is already in the tree to be 
+    decorated by this class is WILL be overwritten. Of course, bootstrap values 
+    will remain untouched.'''
+
+    def __init__(self, **kwargs):
+        '''
+        Parameters
+        ----------
+        reference_tree_path: str
+            Path to the file containing the reference tree, which is used to
+            retroot the tree tree provided to tree
+        old_tree_path: str
+            Path to the file containing the tree to be re-rooted. This tree will
+            be rerooted at the same position as the tree porovided to the 
+            reference_tree
+        '''
+        reference_tree_path = kwargs.pop('reference_tree_path', None)
+        tree_path = kwargs.pop('tree_path', None)
+        
+        if tree_path == None:
```

Comment:
This is in general handled by argparse, no? If it is only checking for programming errors, use
```
kwargs.pop('tree_path')
```
above without the `None` and then it'll just throw a KeyError if it doesn't find it.